src/text_secure_v3/mod.rs

```diff
@@ -118,76 +124,65 @@ impl axolotl::Axolotl for TextSecureV3{
 
     type Mac = hmac::MacResult;
 
-    /// Returns initial Root and Chain keys derived from initial the TripleDH handshake. 
+    /// Returns initial Root and Chain keys derived from initial the TripleDH handshake.
     fn derive_initial_root_key_and_chain_key(
-        local_identity_remote_handshake_dh_secret : &DHShared<Self::IdentityKey>, 
-        local_handshake_remote_identity_dh_secred : &DHShared<Self::IdentityKey>, 
-        local_handshake_remote_handshake_dh_secret : &DHShared<Self::IdentityKey>) -> (Self::RootKey, Self::ChainKey){
-        
-        let master_key : Vec<u8> = [  local_identity_remote_handshake_dh_secret,
-                            local_handshake_remote_identity_dh_secred,
-                            local_handshake_remote_handshake_dh_secret]
-                            .iter()
-                            .flat_map(|x| {x.to_bytes()})
-                            .map(|x|{*x})
-                            .collect();
-
-        let (rk, ck) = keys_from_kdf(&master_key[..], "WhisperText".as_bytes(),&SEED_NULL);
+        local_identity_remote_handshake_dh_secret : &DHShared<Self::IdentityKey>,
+        local_handshake_remote_identity_dh_secret : &DHShared<Self::IdentityKey>,
+        local_handshake_remote_handshake_dh_secret : &DHShared<Self::IdentityKey>)
+            -> (Self::RootKey, Self::ChainKey) {
+
+        let secrets :&[&[u8]] = &[local_identity_remote_handshake_dh_secret.to_bytes(),
```

Comment:
this code is much better than what was there before, however the variable naming deviates from the spec. Would be great for master_key to be referenced for clarity